composer/trainer/trainer.py

```diff
@@ -1691,6 +1766,40 @@ def _train_loop(self) -> None:
         self.engine.run_event(Event.FIT_END)
         self._run_evaluators(Event.FIT_END, log_level=LogLevel.FIT)
 
+    def _eval_train_metrics(self):
+        assert self._train_data_spec is not None, 'The train data spec should be set on __init__ or fit()'
+        assert self.train_metrics is not None, 'The train metrics should be set on __init__ or fit()'
+
+        self.state.model.eval()
+        with torch.no_grad():
+            # Cache the device batch, because `self.state.batch` gets overridden in microbatching loop
+            device_batch = self.state.batch
+
+            # Retry until we successfully complete evaluation
+            while True:
+                # Note: We use uint8 instead of bool as BOR is not supported on all torch.distributed backends
```

Comment:
```suggestion
```

Reply:
Can you explain why? I think it's a useful comment

Reply:
Generally favor terse comments, unless its a significant (and unexpected) bit of code. Suggest moving this to an actual check inside the whichever helper does the reduction, or even just convert the bool -> uint8.

If the comment has to be here, suggest something more terse:
```
found_cuda_oom = 0  # int since bool not supported on all torch.distributed backends
```